### PR TITLE
Code style and sanitization

### PR DESCRIPTION
**Pull request by joshkadis**
And some unit tests for saving and retrieving data.

### CODE REVIEW

File: modules/class-simplechart-save.php [hunk 1]
```diff
@@ -53,10 +48,71 @@ function save_post_action( $post_id ) {
...
+		}
+
+		// handle base64 image string if provided
+		if ( ! empty( $_POST['save-previewImg'] ) ) {
+			$this->_save_chart_image( $post, sanitize_text_field( wp_unslash( $_POST['save-previewImg'] ) ), $this->_default_img_type );
+		}
+
+		// handle raw CSV multiline data
+		if ( ! empty( $_POST['save-rawData'] ) ) {
+			add_filter( 'sanitize_text_field', array( $this, 'sanitize_raw_data' ), 99, 2 );
```
Comment (joshkadis) [inline modules/class-simplechart-save.php:73]: @bcampeau Here's how I ended up handling `sanitize_text_field()` for multiline input

Comment (netaustin) [inline modules/class-simplechart-save.php:73]: i read that exchange in `lead-devs` and am reviewing thusly

File: modules/class-simplechart-save.php [hunk 2]
```diff
@@ -53,10 +48,71 @@ function save_post_action( $post_id ) {
...
+	 */
+	public function sanitize_raw_data( $filtered, $initial ) {
+		// strip newlines from initial input
+		$initial_stripped = preg_replace( '/\n+/', ' ', $initial );
+		if ( $filtered === $initial_stripped ) {
+			// if that's the only difference, then the initial string is fine
+			return $initial;
+		} else {
+			// if anything else was removed by sanitize_text_field(), return an empty string
+			return '';
```
Comment (netaustin) [inline modules/class-simplechart-save.php:113]: might you want an error here? i could see someone (i.e. any reporter) trying to include an HTML tag in their data and being 😩  and 😠 when it fails

File: simplechart.php [hunk 3]
```diff
@@ -28,17 +28,21 @@ class Simplechart {
 	);
 
 	// startup
-	private function __construct(){
+	private function __construct() {
 		// Some stuff we handle different on VIP CLassic™ vs VIP Go and self-hosted sites
 		// thanks matt :)
```
Comment (netaustin) [inline simplechart.php]: you really have to be more specific

File: templates/iframe.php [hunk 4]
```diff
@@ -1,100 +1,101 @@
...
+				}
+			}
+
+			/**
+			 * Mock postMessage from parent window
+			 */
+			function mockParentMessage() {
+				window.postMessage(mockPostMessageData, '*');
+			}
+		</script>
```
Comment (netaustin) [inline templates/iframe.php:98]: I'm not quite clear on the WP Javascript style, but I think it also uses spaces around parens, etc.

**Comment by netaustin** [comment]
some minor style questions / feedback, one minorly substantive request for a literate error message. no need to re-review. 🎈

**Pull request merged**